fix: Provider used after beeing disposed in multi edit page

M123-dev opened the pull request:
### What
Fixes #3357

It's not a perfect provider implementation but that was not the goal. The provider is now only used for disposing, the helpers are passed down manually.

M123-dev committed:
fix: Provider used after beeing disposed in multi edit page

monsieurtanuki reviewed (approved):
Hi @M123-dev!
When I had a look at this issue days ago, it looked like the problem was more with the controllers than with the helpers. I would have rewritten the code for controllers the same way as you did for helpers, but if your code does solve the issue, that's perfect!
I believe there's now a useless `helper` static method - if it's indeed useless please get rid of it, for code clarity.

monsieurtanuki committed:
Merge branch 'develop' into fix-multi-edit-mode

M123-dev commented:
Some of the TextEditingController's listen to the provider, some not thats why I left them untuched. this can be changed in another PR, here the priority was for me to solve the problem.

I have removed the static method, there was still one access to it which I have now also changed it to parameter passing.

M123-dev commented:
Also thanks for your review @monsieurtanuki
